api/storage_service.cc

```diff
@@ -501,13 +501,19 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         if (column_families.empty()) {
             column_families = map_keys(ctx.db.local().find_keyspace(keyspace).metadata().get()->cf_meta_data());
         }
-        return ctx.db.invoke_on_all([keyspace, column_families] (database& db) {
+        auto offstrategy = req_param<bool>(*req, "offstrategy", false);
+        return ctx.db.invoke_on_all([keyspace, column_families, offstrategy] (database& db) {
             std::vector<column_family*> column_families_vec;
             for (auto cf : column_families) {
                 column_families_vec.push_back(&db.find_column_family(keyspace, cf));
             }
-            return parallel_for_each(column_families_vec, [] (column_family* cf) {
+            return parallel_for_each(column_families_vec, [offstrategy] (column_family* cf) {
+                if (!offstrategy) {
                     return cf->compact_all_sstables();
+                } else if (cf->is_pending_offstrategy_compaction()) {
```

Comment:
don't think we need cf->is_pending_offstrategy_compaction(). the same way we don't check first if we need major. we just submit a request to compaction manager, which will then execute it while guaranteeing serialization. if there's nothing to be offstrategy compacted, table will handle it just fine.

Reply:
Fixed in v3